pkg/reconciler/instances/istio/ingress-gateway/config.go

```diff
@@ -0,0 +1,54 @@
+package ingressgateway
+
+import (
+	"context"
+
+	"github.com/go-yaml/yaml"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type NumTrustedProxies *int
+
+const (
+	CMNamespace string = "istio-system"
+	CMName      string = "istio"
+)
+
+// GetNumTrustedProxyFromIstioCM fetches current cluster configuration for "numTrustedProxies" from istio-system/istio ConfigMap.
+// Returns nil if no configuration is present (default configuration).
+func GetNumTrustedProxyFromIstioCM(ctx context.Context, client client.Client) (NumTrustedProxies, error) {
```

Comment:
In the cni implementation we read the actual config [from the istio operator](https://github.com/kyma-incubator/reconciler/blob/c1d7dbc43fa6c6f05896c7025e4344067c2452e7/pkg/reconciler/instances/istio/cni/cni.go#L62). For this implementation we now read from the CM directly. Is there a benefit or drawback of reading from the operator spec? We should agree on a pattern and use it consistently in the future.

Reply:
Oh I missed that there is a bug. I agree that we should add info about the bug, so we know why it's done that way.

Reply:
@triffer @cnvergence Does this comment look OK?
```suggestion
// The desired way to do get the configuration would be to fetch it from Istio Operator, but at this point in time
// it doesn't reflect the actual state of the cluster. This should change when the functionality will be migrated
// to Istio Manager, as reconciliation will run on every update of Istio Custom Resource
func GetNumTrustedProxyFromIstioCM(ctx context.Context, client client.Client) (NumTrustedProxies, error) {
```

Reply:
lgtm, thanks 👍